luaui/barwidgets.lua

```diff
@@ -314,59 +314,48 @@ local zipOnly = {
 	["Widget Profiler"] = true,
 }
 
+local function loadWidgetFiles(vfsMode)
+	local fromZip = vfsMode ~= VFS.RAW
+	local widgetFiles = VFS.DirList(WIDGET_DIRNAME, "*.lua", vfsMode)
+
+	for _, subDirectory in ipairs( VFS.SubDirs(WIDGET_DIRNAME) ) do
+		table.append( widgetFiles, VFS.DirList(subDirectory, "*.lua", vfsMode) )
+	end
+
+	for _, file in ipairs(widgetFiles) do
+		local widget = widgetHandler:LoadWidget(file, fromZip)
+		local excludeWidget = not fromZip and zipOnly[widget.whInfo.name]
+
+		if widget and not excludeWidget then
+			table.insert(unsortedWidgets, widget)
+			Yield()
+		end
+	end
+end
+
 function widgetHandler:Initialize()
 	widgetHandler:CreateQueuedReorderFuncs()
 	widgetHandler:HookReorderSpecialFuncs()
 	self:LoadConfigData()
 
-	-- do we allow userland widgets?
 	if self.allowUserWidgets == nil then
 		self.allowUserWidgets = true
 	end
-	if self.allowUserWidgets and allowuserwidgets then
-		Spring.Echo("LuaUI: Allowing User Widgets")
-	else
-		Spring.Echo("LuaUI: Disallowing User Widgets")
-	end
 
-	-- create the "LuaUI/Config" directory
 	Spring.CreateDir(LUAUI_DIRNAME .. 'Config')
 
-	local unsortedWidgets = {}
+	unsortedWidgets = {}
 
-	-- stuff the raw widgets into unsortedWidgets
 	if self.allowUserWidgets and allowuserwidgets then
-		local widgetFiles = VFS.DirList(WIDGET_DIRNAME, "*.lua", VFS.RAW)
-		for k, wf in ipairs(widgetFiles) do
-			local widget = self:LoadWidget(wf, false)
-			if widget and not zipOnly[widget.whInfo.name] then
-				table.insert(unsortedWidgets, widget)
-				Yield()
-			end
-		end
-	end
-
-	-- stuff the zip widgets into unsortedWidgets
-	local widgetFiles = VFS.DirList(WIDGET_DIRNAME, "*.lua", VFS.ZIP)
-	for k, wf in ipairs(widgetFiles) do
-		local widget = self:LoadWidget(wf, true)
-		if widget then
-			table.insert(unsortedWidgets, widget)
-			Yield()
-		end
+		Spring.Echo("LuaUI: Allowing User Widgets")
+		loadWidgetFiles(VFS.RAW)
+	else
+		Spring.Echo("LuaUI: Disallowing User Widgets")
 	end
 
-	-- stuff the map widgets into unsortedWidgets
-	local widgetFiles = VFS.DirList(WIDGET_DIRNAME_MAP, "*.lua", VFS.MAP)
-	for k, wf in ipairs(widgetFiles) do
-		local widget = self:LoadWidget(wf, true)
-		if widget then
-			table.insert(unsortedWidgets, widget)
-			Yield()
-		end
-	end
+	loadWidgetFiles(VFS.ZIP)
+	loadWidgetFiles(VFS.MAP)
```

Comment:
MAP is part of ZIP so this may be redundant. Previously it used WIDGET_DIRNAME_MAP but it looks like that's the same folder